Multistore: assign a color to a shop group

matthieu-rolland opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | This PR adds a colorpicker in the shop group creation/edition page (multistore context), it saves the color in DB (the ps_shop table) on save.
| Type?         | new feature
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #19303
| How to test?  | 1/ Activate multistore (BO > Shop Parameters > General Parameters)<br>2/ Add a shop group (BO > Advanced Parameters > Multistore > Add Shop group)<br>3/ Check that the colorpicker is here, and that the color is saved on creation and edition.

<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/20125)
<!-- Reviewable:end -->

matthieu-rolland committed:
add color field to ShopGroup entity

matthieu-rolland committed:
add colour picker to shop group BO page

matthieu-rolland committed:
fix wrong column name

matthieu-rolland committed:
cs fixer

matthieu-rolland commented:
I removed the "QA" label since the wording has not been validated yet ;)

Junebyun commented on controllers/admin/AdminShopGroupController.php:
```diff
@@ -187,6 +187,15 @@ public function renderForm()
+                    'label' => $this->trans('Color', [], 'Admin.Catalog.Feature'),
+                    'name' => 'color',
+                    'desc' => [
+                        $this->trans('It will only apply to this group of shops, each store keeps its individual color.', [], 'Admin.Shopparameters.Feature'),
```
Hi @matthieu-rolland ! When you say "_it will only apply to this group of shops_" can you confirm that "**_it_**" means the **color**? 

So that it basically means "_**This color** will only be applied to this group of shops, each store will keep its individual color_"?

Junebyun replied:
```suggestion
                        $this->trans('It will only be applied to this group of shops, each store will keep its individual color.', [], 'Admin.Shopparameters.Feature'),
```

Junebyun commented on controllers/admin/AdminShopGroupController.php:
```diff
@@ -187,6 +187,15 @@ public function renderForm()
+                    'label' => $this->trans('Color', [], 'Admin.Catalog.Feature'),
+                    'name' => 'color',
+                    'desc' => [
+                        $this->trans('It will only apply to this group of shops, each store keeps its individual color.', [], 'Admin.Shopparameters.Feature'),
```
hi @marionf ! Does this sentence have to be part of the tooltip message or as an independent message?

marionf replied:
Hey @Junebyun 
This message isn't part of the tooltip, it's an independant message displayed under the field

matthieu-rolland committed:
wording

Co-authored-by: Junebyun <64894223+Junebyun@users.noreply.github.com>

kpodemski commented:
I don't see any changes in db_structure and update files, am i missing something? or is it still WIP?

NeOMakinG commented:
> I don't see any changes in db_structure and update files, am i missing something? or is it still WIP?

Same question, about this https://github.com/PrestaShop/PrestaShop/pull/20106/files#diff-e70231de5df32627ab34c9619f21e787L17

This PR doesn't require some db changes ?

matthieu-rolland commented:
> > I don't see any changes in db_structure and update files, am i missing something? or is it still WIP?
> 
> Same question, about this https://github.com/PrestaShop/PrestaShop/pull/20106/files#diff-e70231de5df32627ab34c9619f21e787L17
> 
> This PR doesn't require some db changes ?

@NeOMakinG @kpodemski 

The shop_group table is managed by doctrine, I added the color field in ShopGroup.php:

```php
     **
     * @var string
     *
     * @ORM\Column(name="color", type="string", length=50)
     */
    private $color;
```
Doctrine will manage the creation or update at the install and upgrade of the shop.

(Consequently, I removed the modification in the upgrade SQL file in the other PR, @NeOMakinG )

khouloudbelguith commented:
Hi @matthieu-rolland,

Can you please check the conflicts.

Thanks!

matthieu-rolland commented:
PR rebased 👍

khouloudbelguith commented:
Hi @matthieu-rolland,

Can you please check the conflicts.
```
Conflicting files
install-dev/upgrade/sql/1.7.8.0.sql
```

Thanks!

matthieu-rolland commented:
@khouloudbelguith done !

khouloudbelguith commented:
Hi @matthieu-rolland,

I have an error in the console when creating a shop Group.
Steps to reproduce the issue:
1. Enable Multishop context
2. Go to BO > Multistore page
3. Click on "Add a new shop Group"
4. Click on Save
5. see the error

![image](https://user-images.githubusercontent.com/16067358/96102363-beb11a80-0ed6-11eb-9048-4e79d0f048f2.png)


Thanks!

matthieu-rolland commented:
@khouloudbelguith I have the same error on develop without my PR, can you confirm? Maybe I can create an issue for this console error, and we validate this PR? (this PR is getting old, I keep fixing conflicts on it, the sooner it's merged the better imho)

matthieu-rolland commented:
@khouloudbelguith I just rebased the PR (nothing changed in the code). Also the bug you mentioned was fixed in another PR (since it was not related to this PR) 👍

matthieu-rolland commented:
@khouloudbelguith do you think we can merge this PR or do you think it should be tested again ? (the code didn't change since your test)

khouloudbelguith commented:
HI @matthieu-rolland,

I will make a fast check before merge to be sure :slightly_smiling_face: 

Thanks!

khouloudbelguith commented:
Hi @matthieu-rolland,

It is ok :heavy_check_mark: 
https://drive.google.com/file/d/1MS7IDkzE0E69QMuC4wzc3M3ujetOPfEs/view?usp=sharing

Thanks!

Progi1984 commented:
Thanks @matthieu-rolland & @khouloudbelguith
